sound/soc/sof/ipc4-topology.c

```diff
@@ -2127,6 +2154,12 @@ static int sof_ipc4_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget
 		} else {
 			ida_free(&pipeline_ida, swidget->instance_id);
 		}
+	} else if (kcps_adjust) {
+		ret = sof_ipc4_pipeline_send_kcps(sdev, swidget, kcps_adjust);
```

Comment:
Why not make the sof_ipc4_pipeline_send_kcps() handle the !kcps_adjust internally and then you can have this implemented in a more optimistic way:
```c
if (!ret) {
	ret = sof_ipc4_pipeline_send_kcps(sdev, swidget, kcps_adjust);
	if (ret) {
		dev_err(sdev->dev, "failed to increase KCPS: %d, continuing\n", ret);
		ret = 0;
	} else { /* the error case */ }
}
```

Reply:
I was comparing the two options - where to put the check for 0 KCPS adjustment, and in the end I chose this option. It's a really small addition outside and makes it clear immediately at this level, whereas in the function - you'd be calling it just to return immediately. So, unless you feel strongly about this, I'd keep it as is

Reply:
I don't feel strongly about this, just it would make the caller sites cleaner, less number of `if ()`.